src/main/scala/cognite/spark/v1/DataSetsRelation.scala

```diff
@@ -7,15 +7,14 @@ import com.cognite.sdk.scala.common.WithId
 import com.cognite.sdk.scala.v1.resources.DataSets
 import com.cognite.sdk.scala.v1.{DataSet, DataSetCreate, DataSetFilter, DataSetUpdate, GenericClient}
 import io.scalaland.chimney.Transformer
-import org.apache.spark.sql.sources.{Filter, InsertableRelation}
+import org.apache.spark.sql.sources.Filter
 import org.apache.spark.sql.types.{DataTypes, StructType}
 import org.apache.spark.sql.{Row, SQLContext}
 
 import java.time.Instant
 
 class DataSetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
     extends SdkV1Relation[DataSet, String](config, "datasets")
-    with InsertableRelation
```

Comment:
This is also kind of confusing: The `DataSetsRelation` _does_ support writing, but does not implement `InsertableRelation`. I guess the `DefaultSource` for writing completely bypasses the `InsertableRelation` interface and uses `WritableRelation` instead?

Reply:
Yep, I don't yet have full insight into DefaultSource vs InsertableRelation implementations